databricks/koalas/frame.py

```diff
@@ -1915,10 +1910,27 @@ def rename(index):
                      index_name if index_name is not None else rename(index_name)))
                 index_map.remove(info)
 
+        new_data_columns = [
```

Comment:
These codes are needed because previously it removed the index when `reset_index`. Now it sets the default index.